### ISSUE TITLE
[Idea] Opinionated rule that requires `as` to be set for `<Heading>`

### ISSUE THREAD

**Issue by khiga8** [primer/eslint-plugin-primer-react#57]
Currently `<Heading>` defaults to h3 so consumers don't need to explicitly set `as`. 

Default example:
```tsx
<Heading>Some heading</Heading>
```

Should heading `as` always be set to ensure consumers are intentional, rather than obscuring by default?

Here is an example with the same default heading but with level explicitly set:
```tsx
<Heading as="h3">Some heading</Heading>
```

cc: @TylerJDev

**Comment by TylerJDev** [comment]
I think this is a great idea. Since removing the `as` prop entirely would result in a breaking change, I think that this is a good way to slowly lower the amount of usages where the heading level isn't defined. 

We could also add to this rule to ensure that usage of `as` is only being used with heading elements. We already have some safeguards around this, but I think adding this as a rule could supplement those safeguards.

**Comment by TylerJDev** [comment]
Closing as [`a11y-explicit-heading`](https://github.com/primer/eslint-plugin-primer-react/pull/66) rule has been added to `eslint-plugin-primer-react`!
